src/HotChocolate/Core/src/Types/Utilities/Serialization/InputObjectConstructorResolver.cs

```diff
@@ -110,8 +110,22 @@ internal static class InputObjectConstructorResolver
         string name = parameter.Name!;
         string alternativeName = GetAlternativeParameterName(parameter.Name!);
 
-        return (fields.TryGetValue(alternativeName, out field) ||
-            fields.TryGetValue(name, out field));
+        if (fields.TryGetValue(alternativeName, out field) ||
+            fields.TryGetValue(name, out field))
+        {
+            return true;
+        }
+
+        var key = fields.Keys.SingleOrDefault(key => key.Equals(name,
```

Comment:
Should we throw a specific exception to explain that we haven't been able to match the field as there are multple variations that match?

Reply:
Let me check... It could be that a scheme exception would be the right thing here. But let me first check.

Reply:
@michaelstaib  I did consider converting it to a case insensitive dictionary lookup, we'd have to accept that you could not have public properties that differ by case only.  It would certainly be more elegant and performant.

However it is potentially a breaking change.

Reply:
@michaelstaib I've made the field dictonary case-insensitive.  Let me know if you need any changes.